include/jlcxx/module.hpp

```diff
@@ -1188,23 +1188,21 @@ TypeWrapper<T> Module::add_type_internal(const std::string& name, JLSuperT* supe
   protect_from_gc(box_dt);
 
   // Register the type
-  if(!is_parametric)
+  if(is_parametric)
+  {
+    set_const(name, (jl_value_t*)base_dt->name->wrapper);
+    set_const(allocname, (jl_value_t*)box_dt->name->wrapper);
```

Comment:
It seems Windows MSVC really needs this:

```c++
set_const(name, std::forward<jl_value_t*>(base_dt->name->wrapper));
set_const(allocname, std::forward<jl_value_t*>(box_dt->name->wrapper));
```

Reply:
will try